Azure/azure-cli
# [Network] Support vpn gateway generation

**Description of PR (Mandatory)**  
This PR will support vpn gateway generation which has detailed information in #11772.

**Testing Guide**  
- Run command `az network vnet-gateway create -g {rg} -n {gw1} --vnet {vnet1_id} --public-ip-address {ip1} --vpn-gateway-generation Generation1` and check if the command response has vpnGatewayGeneration with `Generation1`

**History Notes:**  
(Fill in the following template if multiple notes are needed, otherwise PR title will be used for history note.)

[Component Name 1] (BREAKING CHANGE:) (az command:) make some customer-facing change.  
[Component Name 2] (BREAKING CHANGE:) (az command:) make some customer-facing change.

---

This checklist is used to make sure that common guidelines for a pull request are followed.
